presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -364,18 +376,19 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
         return new PrestoSparkTaskSourceRdd(sparkContext.sc(), taskSourcesByPartitionId);
     }
 
-    private List<ScheduledSplit> getSplits(Session session, TableScanNode tableScan)
+    private List<ScheduledSplit> getSplits(TableScanNode tableScan, Map<PlanNodeId, SplitSource> splitSources)
```

Comment:
Can we pass in the `splitSource` for the given table scan node and make sure it's closed before leave the method?

Reply:
Unfortunately the `TableScanNode tableScan` parameter cannot be removed from the signature as the `new ScheduledSplit(sequenceId++, tableScan.getId(), split)` requires the id of the TableScan node. It is possible to pass both, `TableScanNode tableScan, SplitSource splitSource` but then it is a little non intuitive to close something that is passed as a parameter, so the close logic must be handled outside. I would prefer to keep it as is. What do you think?

Reply:
@arhimondr : 

>  but then it is a little non intuitive to close something that is passed as a parameter

What about `getSplitsAndCloseSource` ? 😃 

Otherwise, this method will pass in all the split sources (but just get split from one of them) , looks unusual in practice.